### DISCUSSION TITLE
Hello guys, can anyone help with that I need to make self-assign roles like that in my server / I've searched a lot of resources it's only about reaction roles and I don't wanna it so any recommendation?

### DISCUSSION THREAD

**Discussion by co9e-afk** [discordjs/discord.js#8151] [Q&A] [answered]
<img width="734" alt="Screen Shot 2022-06-22 at 4 46 45 PM" src="https://user-images.githubusercontent.com/80815182/175059069-1fa228b4-9c95-460d-a627-380e69a6c9d1.png">

**Comment by vvito7** [comment, marked as answer]
You could read the [guide](https://discordjs.guide/interactions/buttons) about buttons and do it.
In this case you probably won't reply to an interaction, but instead you will create a message with buttons, then listen for button interactions by `interactionCreate` event, then filter it with `<Interaction>.isButton()`. If that makes sense, if not, I highly recommend the guide.

**Reply by co9e-afk** [reply]
where should I write that code 
?
and if there's a tutorial should I follow it would be great

**Reply by vvito7** [reply]
See the Introduction page in the guide
